util/types/datum_test.go

```diff
@@ -184,6 +184,52 @@ func (ts *testTypeConvertSuite) TestToFloat32(c *C) {
 	c.Assert(converted.GetFloat64(), Equals, datum.GetFloat64())
 }
 
+// mustParseTime is similar to ParseTime but panic if any error occurs.
+func mustParseTimeIntoDatum(s string, tp byte, fsp int) (d Datum) {
+	t, err := ParseTime(s, tp, fsp)
+	if err != nil {
+		panic("ParseTime fail")
+	}
+	d.SetMysqlTime(t)
+	return
+}
+
+func (ts *testDatumSuite) TestToJSON(c *C) {
+	ft := NewFieldType(mysql.TypeJSON)
+	sc := new(variable.StatementContext)
+	tests := []struct {
+		datum    Datum
+		expected string
+		success  bool
+	}{
+		{NewIntDatum(1), `1.0`, true},
+		{NewFloat64Datum(2), `2`, true},
+		{NewStringDatum("\"hello, 世界\""), `"hello, 世界"`, true},
+		{NewStringDatum("[1, 2, 3]"), `[1, 2, 3]`, true},
+		{NewStringDatum("{}"), `{}`, true},
+		{NewIntDatum(1), `true`, true},
+		{mustParseTimeIntoDatum("2011-11-10 11:11:11", mysql.TypeTimestamp, 0), `"2011-11-10 11:11:11"`, true},
+
+		{NewStringDatum("hello, 世界"), "", false},
+	}
+	for _, tt := range tests {
+		obtain, err := tt.datum.ConvertTo(sc, ft)
```

Comment:
I prefer keep the if-else, because I can compare obtained and expected datum only if success == true.